### ISSUE TITLE
[Perf] Higgs TTS: batched audio encoder

### ISSUE THREAD

**Issue by xinlij** [sgl-project/sglang-omni#603]
## Motivation

The audio encoder stage (`create_audio_encoder_executor`) collects up to `max_batch_size=8` requests via `SimpleScheduler`, but no `batch_compute_fn` is provided — each request is still encoded serially with a single `codec.encode_reference` call on a `[1, 1, L]` waveform.

`HiggsAudioV2TokenizerModel.encode()` already accepts batched input `(batch_size, channels, num_samples)` and returns `(batch_size, num_quantizers, codes_length)` (see `_vendored/higgs_audio_v2_tokenizer_hf.py` line 789). Stacking same-length waveforms into `[B, 1, L]` and running a single forward pass would reduce per-request encoder latency at concurrency ≥ 2 — the same gap that #569 / PR #574 closed for the vocoder.

## Current path

```python
# HiggsAudioCodec.encode_reference — single waveform at a time
wav = _to_mono_3d(waveform)                          # [1, 1, L]
codes_BNT = self.model.encode(wav).audio_codes       # [1, N, T]
return codes_BNT.squeeze(0).transpose(0, 1).cpu()    # [T, N]

# stages.py — _encode()
ref_codes_TN = codec.encode_reference(waveform, sample_rate=24000)  # one at a time
```

`SimpleScheduler` is instantiated with `max_batch_size=8` but without `batch_compute_fn`, so batch collection is unused.

## Proposed change

1. Add `encode_batch` to `HiggsAudioCodec`, using **bucketing by waveform length** rather than zero-padding to `L_max`.

The acoustic encoder uses symmetric Conv1d padding (`padding = (kernel - 1) // 2`), making it non-causal — zero-padding shorter waveforms to a longer `L_max` corrupts codes near the boundary, the same issue PR #574 encountered on the decoder side. Bucketing avoids this entirely: only same-length waveforms are stacked into a single forward pass; items with different lengths are processed in separate per-bucket calls.

```python
def encode_batch(self, waveforms: list[torch.Tensor]) -> list[torch.Tensor]:
    """Encode a batch of [1, 1, L_i] waveforms → list of [T_i, N] code tensors."""
    from collections import defaultdict
    buckets: dict[int, list[int]] = defaultdict(list)
    for i, w in enumerate(waveforms):
        buckets[w.shape[-1]].append(i)

    results = [None] * len(waveforms)
    for length, indices in buckets.items():
        if len(indices) == 1:
            results[indices[0]] = self.encode_reference(waveforms[indices[0]])
            continue
        batch = torch.cat([waveforms[i] for i in indices])  # [B, 1, L]
        batch = batch.to(device=self.device, dtype=self._dtype)
        codes_BNT = self.model.encode(batch).audio_codes    # [B, N, T]
        for j, idx in enumerate(indices):
            results[idx] = codes_BNT[j].transpose(0, 1).to(torch.long).cpu()
    return results
```

2. Add `_encode_batch` in `stages.py` and wire it into `SimpleScheduler` as `batch_compute_fn`, mirroring the `_vocode_batch` pattern from PR #574.

## Scope

- `sglang_omni/models/higgs_tts/audio_codec.py` — add `encode_batch`
- `sglang_omni/models/higgs_tts/stages.py` — add `_encode_batch`, wire `batch_compute_fn`
- `tests/unit_test/higgs_tts/` — numerical parity test for batched vs single encode

## Acceptance Criteria

**Correctness**
- [ ] `encode_batch([waveform])` produces bit-identical output to `encode_reference(waveform)` for single-item batches
- [ ] Multi-item batches with same-length waveforms produce identical codes to individual `encode_reference` calls
- [ ] Existing unit tests under `tests/unit_test/higgs_tts/` still pass

**Performance**
- [ ] Encoder stage throughput strictly higher than single-encode baseline at concurrency ≥ 2
- [ ] No regression at concurrency = 1

## Known Pitfalls

**Non-causal encoder requires bucketing, not padding.** The acoustic encoder uses symmetric Conv1d padding and is therefore non-causal. Zero-padding shorter waveforms to `L_max` would corrupt codes near the padding boundary. Bucketing by waveform length avoids this entirely, at the cost of reducing effective batch size when concurrent requests have different-length reference clips.

**Minimum length requirement.** `encode_reference` zero-pads clips shorter than 1 s (`SAMPLE_RATE` samples) because the codec errors otherwise. `encode_batch` must apply the same minimum-length padding to each waveform before stacking.

## References

- Batched vocoder pattern: `sglang_omni/models/higgs_tts/stages.py` `_vocode_batch` (PR #574)
- `SimpleScheduler` batch API: `batch_compute_fn` parameter
- `HiggsAudioV2TokenizerModel.encode` batched signature: `_vendored/higgs_audio_v2_tokenizer_hf.py` line 789
- Tracking issue: #478
